cromulent/model.py

```diff
@@ -812,10 +810,78 @@ def _linked_art_boundary_okay(self, top, prop, value):
 		# Without replacement, just return True always
 		return True
 
+	def list_all_props(self, filter=None):
+		props = []
+		for c in self._classhier:		
+			for k,v in c._all_properties.items():
+				if not k in props and (filter is None or isinstance(filter, v.range) or filter is v.range):
+					props.append(k)
+		return props
+
+	def list_my_props(self, filter=None):
+		d = self.__dict__.copy()		
+		props = []
```

Comment:
Could this be a generator instead of collecting properties into an array and returning it at the end?

Reply:
It _could_ be a generator ... but the number of properties doesn't seem worthwhile to use the generator pattern, rather than just creating the list?

Will create an issue to discuss generators vs materialized lists.